util/java/tmxviewer-java/src/main/java/TMXViewer.java

```diff
@@ -119,10 +132,12 @@ public void paintComponent(Graphics g) {
         g2d.fill(clip);
 
         // Draw each tile map layer
-        for (MapLayer layer : map) {
-            if (layer instanceof TileLayer) {
-                renderer.paintTileLayer(g2d, (TileLayer) layer);
-            }
+        for (TileLayer layer : map.getLayers()) {
+            renderer.paintTileLayer(g2d, layer);
+        }
+        // Draw each object group
+        for (ObjectGroup group : map.getObjectGroups()) {
```

Comment:
Hm, this is not how layers work, since layers of different types are supposed to get interleaved. Would it still be possible to set it up that way?